internal/consensus/state.go

```diff
@@ -875,7 +882,11 @@ func (cs *State) handleMsg(ctx context.Context, mi msgInfo) {
 		// if the proposal is complete, we'll enterPrevote or tryFinalizeCommit
 		added, err = cs.addProposalBlockPart(ctx, msg, peerID)
 		if added {
-			cs.statsMsgQueue <- mi
+			select {
+			case cs.statsMsgQueue <- mi:
```

Comment:
this channel isn't closed as far as I can tell. 

(which I think is fine? I tend think that not closing channels is fine if you know all of the users of the channel are going to exit, then it can just get GC'd eventually?)

Reply:
Yeah, as long as there are no readers or writers blocked on the channel, it doesn't matter if it gets closed.